test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/ViewModels/ReadmePreviewViewModelTests.cs

```diff
@@ -33,50 +33,134 @@ public void Constructor_Defaults()
+            var target = new ReadmePreviewViewModel(mockFileService.Object, ItemFilter.Installed);
+            var package = new DetailedPackageMetadata();
+            package.ReadmeFileUrl = "C://path/to/readme.md";
+            await target.SetPackageMetadataAsync(package, CancellationToken.None);
 
             //Act
-            await target.LoadReadmeAsync("C://path/to/readme.md", CancellationToken.None);
+            await target.SetCurrentFilterAsync(ItemFilter.All);
+
+            //Assert
+            Assert.False(target.ErrorLoadingReadme);
+            Assert.Equal(string.Empty, target.ReadmeMarkdown);
+        }
+
+        [Theory]
+        [InlineData(ItemFilter.Installed)]
+        [InlineData(ItemFilter.Consolidate)]
+        [InlineData(ItemFilter.UpdatesAvailable)]
+        public async Task SetCurrentFilter_ItemFilterRenderingLocalReadme_ReadmeReturned(ItemFilter filter)
+        {
+            //Arrange
+            var readmeContents = "readme contents";
+            using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(readmeContents));
+            var mockFileService = new Mock<INuGetPackageFileService>();
+            mockFileService.Setup(x => x.GetReadmeAsync(It.IsAny<Uri>(), It.IsAny<CancellationToken>())).ReturnsAsync(stream);
+            var target = new ReadmePreviewViewModel(mockFileService.Object, ItemFilter.All);
+            var package = new DetailedPackageMetadata();
+            package.ReadmeFileUrl = "C://path/to/readme.md";
+            await target.SetPackageMetadataAsync(package, CancellationToken.None);
+
+            //Act
+            await target.SetCurrentFilterAsync(filter);
+
+            //Assert
+            Assert.False(target.ErrorLoadingReadme);
+            Assert.Equal(readmeContents, target.ReadmeMarkdown);
+        }
+
+        [Fact]
+        public async Task SetPackageMetadataAsync_WithLocalReadmeUrl_RenderLocalReadmeTrue_ReadmeReturned()
+        {
+            //Arrange
+            var readmeContents = "readme contents";
+            using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(readmeContents));
+            var mockFileService = new Mock<INuGetPackageFileService>();
+            mockFileService.Setup(x => x.GetReadmeAsync(It.IsAny<Uri>(), It.IsAny<CancellationToken>())).ReturnsAsync(stream);
+            var target = new ReadmePreviewViewModel(mockFileService.Object, ItemFilter.Installed);
+            var package = new DetailedPackageMetadata();
+            package.ReadmeFileUrl = "C://path/to/readme.md";
+
+            //Act
+            await target.SetPackageMetadataAsync(package, CancellationToken.None);
 
             //Assert
             Assert.False(target.ErrorLoadingReadme);
             Assert.Equal(readmeContents, target.ReadmeMarkdown);
-            Assert.True(target.CanDetermineReadmeDefined);
+        }
+
+        [Fact]
+        public async Task SetPackageMetadataAsync_WithLocalReadmeUrl_RenderLocalReadmeFalse_ReadmeReturned()
```

Comment:
Should it not return a readme or should it be returning the contents from the `stream` ?

```suggestion
        public async Task SetPackageMetadataAsync_WithLocalReadmeUrl_RenderLocalReadmeFalse_EmptyReadmeReturned()
```